File: dropwizard/src/test/java/com/yammer/flopwizard/example/UploadResource.java
package com.yammer.flopwizard.example;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.core.MediaType;

@Path("/upload")
@Consumes(MediaType.WILDCARD)
public class UploadResource {
    private static final Logger LOGGER = LoggerFactory.getLogger(UploadResource.class);
    
    @POST
    public void upload(String body) {
        LOGGER.info("New upload: " + body);
    }
}

File: dropwizard-scala_2.9.1/src/test/scala/com/yammer/dropwizard/examples/UploadResource.scala
package com.yammer.dropwizard.examples

import javax.ws.rs.core.MediaType
import javax.ws.rs.{POST, Consumes, Path}
import com.yammer.dropwizard.Logging
import com.yammer.metrics.annotation.Timed

@Path("/upload")
@Consumes(Array(MediaType.WILDCARD))
class UploadResource extends Logging {
  @POST
  @Timed
  def upload(body: String) {
    logger.info("New upload: %s", body)
  }
}
